src/chart/main.c

```diff
@@ -868,51 +868,77 @@ static char *encode_colorchecker(int num, const double *point, const double **ta
   return dt_exif_xmp_encode_internal((uint8_t *)&params, sizeof(params), NULL, FALSE);
 }
 
+static int compare_L_source(const void *x_, const void *y_) {
+  const double *x = x_;
+  const double *y = y_;
+  return (*x) < (*y) ? -1 : ((*x) > (*y) ? +1 : 0);
+}
+
 static void process_data(dt_lut_t *self, double *target_L, double *target_a, double *target_b,
                          double *colorchecker_Lab, int N, int num_tonecurve, int sparsity)
 {
-  if (num_tonecurve < 2)
-    num_tonecurve = 2;
+  // sort all gray patches according to the L channel of the source colors
+  double *grays = malloc(sizeof(double) * 6 *num_tonecurve);
```

Comment:
```
double *grays = malloc(sizeof(double) * 6 * num_tonecurve);
```

Reply:
These two lines of code are no longer required. Not removing them is a bug.

Reply:
I am unsure if I understand your comment. Please increase verbosity level.

Reply:
```diff
-+  double *grays = malloc(sizeof(double) * 6 *num_tonecurve);
++  double *grays = malloc(sizeof(double) * 6 * num_tonecurve);
```

In other words, what is `*num_tonecurve`?
Is `num_tonecurve` a pointer?
And then you have `6` and "dereferenced" pointer which wouldn't even compile.

TLDR: [.git/hooks/pre-commit](https://github.com/darktable-org/darktable/blob/master/CONTRIBUTING.md#coding-style)

Reply:
I configured `git-clang-format` and revised the code. It should comply with the darktable coding style now.